### PR TITLE
SavePreferences Use Case

### PR DESCRIPTION
**Pull request by samaslanowicz**
Completed all of the classes in the save preferences use case, as well as the controller. Can you guys check this out @merrickliu888 @jackkfan0305 ?

### CODE REVIEW

**Review by jackkfan0305 — COMMENTED** [review]
I think you should change the ```void savePreferences()``` method name in SaveDataAccessInterface to ```void save()```. Since FileDataAccessObject will implement the interface, ```save()``` method will already save user preferences so making another  ```savePreferences()``` method is redundant. Otherwise, everything else looks good!

**Review by merrickliu888 — CHANGES_REQUESTED** [review]
Nice. Just separate the frontend classes to another PR. Also, refactor some of the variables like Jack said in Lab.

File: src/use_case/save_preferences/SavePreferencesDataAccessInterface.java [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package use_case.save_preferences;
+
+public interface SavePreferencesDataAccessInterface {
+    void savePreferences();
```
Comment (merrickliu888) [inline src/use_case/save_preferences/SavePreferencesDataAccessInterface.java:4]: Change the function signature to `void saveUserPreference(String username, UserPreference preference)`

**Review by merrickliu888 — CHANGES_REQUESTED** [review]

**Review by merrickliu888 — APPROVED** [review]
Looks good. Now we just have to test, and if anything comes up we'll fix it. We'll also need to re-evaluate some of the front end things, but that is for another PR.

**Pull request merged**